CONTRIBUTING_NI.md

```diff
@@ -71,3 +71,28 @@ git cherry-pick 095cea2^..d919979
 
 Carefully resolve any merge conflicts. Run `git cherry-pick --skip` for any changes that were accepted upstream since the last version bump.
 Once the cherry pick is completed, force push main (`git push -f`). If you are not an admin, ask one to do this step for you.
+
+If you find you need to update `main` multiple times during the same release upgrade process (you may find issues that need resolution after initial attempt), or as part of the cherry-pick process you inadvertently introduced new commits, you can/should collapse those commits into a single commit (the most recent commit) using `git rebase -i (commit before first new commit)`.
+
+### Tips and Tricks
+
+#### Setup
+
+First, make sure to read and follow the directions of the [Developer Guide](./contribute/developer-guide.md). 
+
+#### Windows users
+
+As of Grafana 11, you should be using a later version of node (e.g. 22), as the noted version in the [linked blog](https://grafana.com/blog/2021/03/03/how-to-set-up-a-grafana-development-environment-on-a-windows-pc-using-wsl/) of 14.15.5 will fail to compile the backend and frontend code. Using a newer version of Go (more recent than the noted 1.15) may also be needed (e.g. 1.23).
```

Comment:
> Using a newer version of Go (more recent than the noted 1.15) may also be needed (e.g. 1.23).

The main problem I ran into was that the version of Go I tried to use (maybe 1.21.x?) didn't work. The only way I was able to compile was using the specific 1.15 version the docs referenced, which was a bit annoying since it is now an archived version.

That being said, what you have here is probably fine since you got grafana to compile. I think I just got unlucky with a broke/incompatible version of Go.